app/build.gradle

```diff
@@ -10,6 +10,7 @@ dependencies {
     implementation 'com.github.nicolas-raoul:Quadtree:ac16ea8035bf07'
     implementation 'fr.avianey.com.viewpagerindicator:library:2.4.1.1@aar'
     implementation 'in.yuvi:http.fluent:1.3'
+    implementation 'com.github.chrisbanes:PhotoView:2.0.0'
```

Comment:
Apache license so no problem :-)

Reply:
I tried to implement the pinch zoom on my own earlier but it had a flaw. While dragging the image the bounds weren't restricted as a result user was able to drag the image out of screen. Debugging it was becoming a tedious task so i used this to add functionality of pinch zoom :)